lib/capybara/node/matchers.rb

```diff
@@ -892,7 +892,7 @@ def _verify_text(type = nil, expected_text, **query_options) # rubocop:disable S
       def _set_query_session_options(*query_args)
         query_args, query_options = query_args.dup, {}
         # query_options = query_args.pop if query_options.empty? && query_args.last.is_a?(Hash)
-        query_options = query_args.pop if query_args.last.is_a?(Hash)
+        query_options = query_args.pop if query_args.last.is_a?(Hash) && query_args.last.keys.none?(String)
```

Comment:
I'm pretty dissatisfied with this solution. Since it's related to positional-to-keyword argument coercion, I wonder if there's a more wholistic approach that could be take to resolve this. For example, this would still fail for symbol keys that aren't filter options:

````ruby
assert_selector :table_row, First: "John", Last: "Doe"
```